Bug/fixes and changes

KarinMoveo opened the pull request:
Hello :)

In this branch, I have completed all the frontend changes and fixes from our last PR:

1. Instead of writing the URL every time, I used process.env.BASE_URL for consistency.

2. In cases where the URL was repeated with the same prefix, such as "http://localhost:3000/chef....", I made it more generic by defining a constant like this:
const BASE_URL = process.env.REACT_APP_BASE_URL;
const chefsBaseUrl = `${BASE_URL}/chefs`;

3. I removed the "mockData" folder from the project to clean up unnecessary files (now I am using the data from DB).

Thank you :)

KarinMoveo committed:
Refactoring API Endpoint URL and Adding Environment Variables

KarinMoveo committed:
Deleting mockData from frontend

KarinMoveo committed:
Fixing components

KarinMoveo committed:
Delete api call

rabeafm commented on src/modules/chefs/api.ts:
```diff
@@ -1,24 +1,27 @@
 export async function deleteChefFromAPI(id: any){
     const result = await axios.delete(
-        `http://localhost:5000/chefs/deleteChef/${id}`
+        `${chefsBaseUrl}/deleteChef/${id}`
```
there is no need to write deletechef, addchef and updatechef since they use different methods (GET,POST,PUT,DELETE, etc..) they can all end with chefs and the difference could be the method, also in get all and get they can be differentiated by parameter

rabeafm commented on src/modules/home/api.ts:
```diff
@@ -1,16 +1,18 @@
+
 export async function getPopularRestaurantsFromAPI(){
-    const result = await axios.get('http://localhost:5000/restaurants/popular');
+    const result = await axios.get(`${BASE_URL}/restaurants/popular`);
```
instead of using different paths for popular, chef of the week and signature dishes you can utilize query parameters

rabeafm commented on src/modules/home/components/SignatureDishesContainer.tsx:
```diff
@@ -17,6 +17,7 @@ function SignatureDishesContainer() {
 		async function getSignatureDishes() {
 			try {
 				const result = await getSignatureDishesFromAPI();
+				console.log(result.data);
```
dont leave console.logs in the code

rabeafm commented on src/modules/restaurants/api.ts:
```diff
@@ -1,21 +1,19 @@
-export async function getRestaurantByIDFromAPI(id: any) {
-	const result = await axios.get(`http://localhost:5000/restaurants/${id}`);
+export async function getRestaurantByNameFromAPI(name: any) {
+	const result = await axios.get(`${BASE_URL}/restaurants/${name}`);
```
even though i said oits ok to use restaurants names, id is the better way since two restaurants could have the same name

rabeafm commented on src/modules/restaurants/components/SpecificRestaurantPage.tsx:
```diff
@@ -54,28 +54,21 @@ function SpecificRestaurantPage() {
 	};
 
 	useEffect(() => {
-		async function getRestaurantByID() {
```
by id is better

rabeafm commented on src/modules/restaurants/components/SpecificRestaurantPage.tsx:
```diff
@@ -54,28 +54,21 @@ function SpecificRestaurantPage() {
-		}
-		getAllDishesFromCategory();
-	}, [selectedMealCategory]);
+		const filteredDishes = restaurant?.dishes.filter((dish) => dish.mealType.includes(selectedMealCategory));
```
We need to think about where we should put the filters?
https://stackoverflow.com/questions/8368002/search-filter-should-i-do-it-client-side-or-server-side

rabeafm commented on src/shared/assets/images/checkout.svg:
why are you deleting this file, isn't it a front asset?

rabeafm commented on src/shared/assets/images/orderHistory.svg:
same remark

rabeafm commented on src/shared/util.ts:
```diff
@@ -1,10 +1,17 @@
 export function isRestaurantOpen(from: string, to: string): boolean {
+    const timeFormatRegex = /^([01]\d|2[0-3]):([0-5]\d)$/; // HH:mm format regex
```
nice

KarinMoveo committed:
Renaming methods, Deleting console.log, changing get restaurant by id instead of name

KarinMoveo committed:
Deleting things that has no use

KarinMoveo commented:
- Things I fixed:
1. Renamed methods (for example, changed "deleteChef" to "chef" when the method is for deleting).
2. Removed console.log statements.
3. Changed "getRestaurantByName" to "getRestaurantById" (working with id instead of name).

- General comments:
1. I've maintained different paths for "popular," "chef of the week," and "signature dishes" because each of them originates from a different route (popular restaurant - from restaurant, chef of the week - from chef, signature dishes - from dishes). I want to avoid confusion as I continue working on the project.
2. I implemented the filter on the frontend because I believed it would be more efficient since I don't have a large amount of data.
3. I removed the checkout and order history images as I have no use for them. Instead, I'm using a general button I created throughout the project.

Thank you! :)

KarinMoveo committed:
Merge branch 'develop' into bug/fixes-and-changes
